---
layout: article
title: <font size=6> 文献分享：Haplotype-resolved de novo assembly using phased assembly graphs with hifiasm </font>
tags: papers
article_header:
  type: cover
  image: 
    src: /post_image/hifiasm/hifiasm.PNG
---

# Foreword（写在前面）
今天给大家分享的这篇文章hifiasm是2021年2月份哈佛的Heng Li发在Nature methods上的一篇关于基因组拼接的新作，Heng Li是生物信息算法这方面的大佬了，BEA-MEM,minimap2,wtdbg2等等，做了很多序列比对、拼接方面的优秀算法。今天这篇讲的是关于单倍型分辨率的从头拼接算法。<!--more-->先做一个简单的介绍，所谓的Haplotype-resolved和phased assembly是一个意思（见Heng Li去年的一篇nature biotech的摘要[Chromosome-scale, haplotype-resolved assembly of human genomes](https://www.nature.com/articles/s41587-020-0711-0)），指的就是我们能够重构出多倍体物种的所有的单倍型。就人类的基因组而言，对于每对同源染色体其实我们有父源和母源两条，上面可能携带了不同的等位基因，但是传统的assembly方法都忽略了两条染色体的差异，其实最后生成的contigs是类似consensus的sequence。其中的原因也比较显然，因为我们测序的时候将所有基因组随即打碎，我们也分辨不出父源母源的序列。如下图，一般我们称左边得到的concensus序列叫做Genotype,而我们的目标是phase出来父源母源两条序列。那么将一对等位基因都分别还原出来，对于遗传学研究、各种疾病检测的意义肯定是不言而喻的。
![传统拼接的consensus序列 VS phased序列](/post_image/hifiasm/phased_seq.png)

__Fig.1 传统拼接的consensus序列 Vs phased序列.__

那么，我们有什么手段可以进行phasing呢？

1.首先，最直接的方法，我们可以再测该个体的母亲和父亲的基因组，当然这个方法缺点就在于我们又多测了两个基因组，而且有时候我们可能都得不到父母的基因组；其次，如果父母都是杂合子AB，那么我们也不能判断出来子代等位基因AB的来源；在2018年，National Institutes of Health (NIH) and the United States Department of Agriculture (USDA)的两个科学家在nature biotech上发布了Trio binning算法（Trio指的就是Father-mather-child三样本），我们可以看一些Heng Li在20年7月发的一篇natrue biotech对trio binning的评价
>
*FALCON-Phase, which extends FALCON-Unzip ...... but it cannot achieve chromosome-long phasing. Trio binning is the only published method that can do this, plus the assembly and phasing of entire chromosomes. ...... However, trio binning is unable to resolve regions heterozygous in all three samples in the trio and will leave such regions unphased. More importantly, parental samples are not always available—for example, for samples caught in the wild or when parents are deceased. For Mendelian diseases, de novo mutations in the offspring will not be captured and phased with the parents if there are no other heterozygotes nearby. This limits the application of trio binning.*
>

~~害，这篇FALCON-Phase在19年4月就挂在biorxiv上了，21年4月28号也就是前几天才终于发在了Nature Communications~~

下面是trio binning的一个workflow，红色表示母源的染色体，蓝色表示父源，然后对于父母的基因组short-read测序，然后我们可以得到k-mers的一个分布情况，也就是b中的韦恩图。根据这个分布，对于子代基因组的long-read进行分类，分为父源、母源以及unassigned，然后分别进行组装。
![Outline of trio binning and haplotype assembly](/post_image/hifiasm/trio_binning.PNG)

__Fig.2 Outline of trio binning and haplotype assembly[^1].__


2.第二种是LD phase，也就是根据连锁不平衡，有些区域的染色体在减数分裂同源染色体重组的时候，更不容易重组，经常连锁在一起，我们也可以构建一些遗传的统计模型，HMM模型来预测我们所测样本的可能单倍型，这个涉及遗传的知识比较多，就不再详述，当然了问题也很多，LD方法对于罕见突变或者个体特有的突变完全无能为力；

3.Physical phase，也就是我们不依赖任何外部的信息，就从该个体的基因组测序结果进行重构，因为从本质上，每一个read一定来自同一个单倍型，所以其实就是一个局部的phase，然后我们的目标是将这些局部的phase延伸成尽可能长的单倍型序列，所以我们最终目标是，将整个姐妹染色单体分别重构，或者说我们想得到telomere-to-telomere的phase。这也是我们这篇文章所做的phasing算法就属于这类方法。

# Background
然后我们先来看几个概念，这几个概念的名词解释来自于前几天Heng Li自己发的Blog [Heng Li的博客](https://lh3.github.io/2021/04/17/concepts-in-phased-assemblies)

**Haplotig**: a contig that comes from the same haplotype. In an unphased assembly, a contig may join alleles from different parental haplotypes in a diploid or polyploid genome.

**Primary assembly**: a (haploid) complete assembly with long stretches of phased blocks.

**Haplotype-resolved assembly**: sets of (haploid) complete assemblies consisting of haplotigs, representing an entire diploid/polyploid genome. This concept has been inconsistently used in publications without a clear definition. The above is my take in our recent papers.

也就是Haplotig表示序列来自同一单倍型的contig，而Haplotype-resolved assembly表示一些(haploid) complete assemblies的集合，也就是这些组装（由haplotigs构成）共同代表了整个二（多）倍体基因组。所谓的Primary assembly就是一个中间阶段吧，由一些已经phased的较长序列组成，但是其实Primary assembly的结构是mosaic的，后文将会再次提到，并详细介绍。

这篇文章使用的是HiFi-read来进行组装的，如果不熟悉三代测序中Pacbio方法中的CCS序列（HiFi-read），可以看下面这个图，根据我的调研，之前官网文档上表示CCS序列长度在500bp-3kb，较新的数据根据下面这个篇文章所述，read平均长度可以达到13.5(kb)，准确度在99.8%。之所以准确度比正常的三代测序要高，是因为其对一个read循环测了多次，因为错误是随机的，所以可以相互矫正.关于CCS(HiFi-read)的详细内容，可以参见这篇19年的nature biotech，是Pacbio公司发表的[Accurate circular consensus long-read sequencing improves variant detection and assembly of a human genome](https://www.nature.com/articles/s41587-019-0217-9).
![HiFi read](/post_image/hifiasm/How-to-get-HiFi-reads_v2.png)

__Fig.3 How to get HiFi-read[^2].__

>
*A great challenge to the assembly of heterozygous samples is caused by the 5–15% sequencing error rate of older long reads. With this high error rate, it is difficult to distinguish errors from heterozygotes occurring at a rate of <0.1% in humans. The recent availability of high-fidelity (HiFi) reads produced by PacBio has changed the equation.Generated from the consensus of multiple sequences of the same DNA molecule, HiFi reads have a much lower error rate of <1%. With HiFi, standard trio binning can produce contigs of 17 Mb.*
>

为什么使用HiFi-read，其实也很好理解，因为同源染色体本身的差距就很小，根据这篇文章所讲，人类的杂合子的差距在0.1%以内，如果read本身误差很大，那么对于回复更精确的同源序列就更是天方夜谭了。
PS：个人认为，可能正是HiFi-read准确度、长度都大大提高，所以才使得单倍型的组装进一步发展，使得染色体规模的单倍型组装成为可能（确实最近这方面的文章在nature子刊上发表了很多）.

# Introduction
## 仅用read进行phasing
>
*Most of them (assemblers) collapse different homologous haplotypes into a consensus representation with heterozygous alleles frequently switching in the consensus. ...... Falcon-Unzip recovers heterozygous alleles by ‘unzipping’ them in an initial collapsed assembly. It produces a pair of assemblies, one primary assembly representing a mosaic of homologous haplotypes, and one alternate assembly composed of short haplotype-specific contigs (haplotigs) for alleles absent from the primary assembly. The alternate assembly is often fragmented and does not represent a complete haplotype, making it less useful in practice.*
>

目前已知的很多三代基因组拼接软件得到是序列都是同源染色体的consensus序列。
Falcon-Unzip选择了从collapsed assembly中“unzip”出等位基因，然后得到两组assembly，一个称为primary assembly，其中的contig是同源的单倍型的序列交错出现的，也就是所谓mosaic的结构，另外一套assembly是一些较短的Haplotig，也就是primary contig中一些基因的相应的等位基因序列如下图所示，所以其实Flacon-Unzip并没有实现整个染色体规模的phased，两个assembly都不代表任何一个单倍型。Falcon-phase是前几天4月28号发在NC上的，在biorxiv上挂了两年，也只是借用了Hi-C数据使得haplotig更长，仍没达到染色体规模。

![Falcon-Unzip output](/post_image/hifiasm/Falcon.PNG)

__Fig.4 Falcon output[^3].__

>
*Trio binning addresses these issues by globally partitioning long reads upfront with parental short reads and then performing two separate assemblies on the partitioned reads. This strategy works well for samples with high heterozygosity,
but for a human sample sequenced with noisy long reads, it only produces fragmented assemblies with ~1.2-Mb contigs.*
>

刚才我们简单介绍了Trio binning的原理，因为其根据k-mers的父源、母源来对子代long-reads进行分类，其实文章用了Angus and Brahman两个品种 的牛（安格斯和婆罗门）做了实杂交实验，再进行测序，但是对于亲缘关系更近的人类，trio binning的contigs长度就大大减小了，因为噪声更多，分辨read属于父母哪一方更困难。因此，组装出来的contigs的连续性就会变差。

## 使用额外技术phasing

>
*Recent works relying on high-throughput chromosomal confirmation capture (Hi-C) or strand sequencing (Strand-seq) read binning can achieve better contiguity and phasing accuracy. These pre-binning algorithms all use short k-mers or short reads to partition HiFi reads. They may not identify haplotype-specific markers in complex regions and result in wrong read partitions, which will negatively affect the assembly, as shown later. In addition, both Hi-C and Strand-seq binning start with a collapsed assembly and have the same issues as Falcon-Unzip.*
>

刚才提到的20年7月Heng Li发表在Nature Biotech的Chromosome-scale, haplotype-resolved assembly of human genomes，就利用了Hi-C染色体结构数据来延长phased的片段；另一种技术称为Strand-seq，大致的原理图如下，首先在细胞复制的时候加入一种叫做BrdU的碱基类似物，这种物质可以在UV射线照射下降解，所以细胞复制新合成的链后面就会被降解，被保留的就是原始细胞的模板链。那么在该细胞细胞分类得到的子代细胞中，就有下面三种类型，WW、CC、WC（我们把模板链的正链记为Watson(W)链，反链记为Crick(C)链）.那么对于WC这种类型的子代细胞，我们进行测序，将read回贴到参考基因组上，很容易就可以分辨出来read属于父源的染色体还是母源的染色体（因为比对在参考基因组W、C链来分别来自两条姐妹染色体）
对于利用Strand-seq来phased基因组的，可以看这两篇文章

[Fully phased human genome assembly without parental data using single-cell strand sequencing and long reads](https://www.nature.com/articles/s41587-020-0719-5#MOESM1)

[Direct chromosome-length haplotyping by single-cell sequencing](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5088598/)

![Strand-seq](/post_image/hifiasm/Strand-seq.PNG)

__Fig.5 Strand-seq[^4].__

但是上述方法，思路都是先将HiFi reads进行分类（依赖k-mers或者short reads），这会导致read分类的错误。**后面补充······**

## unitig graph

# Methods

## Overview of hifiasm

## Haplotype-aware error correction

>
*Hifiasm performs all-versus-all read overlap alignment and then corrects sequencing errors.Given a target read to be corrected, hifiasm inspects the alignment of reads overlapping with the* **target read**.

**Def**: *A position on the target read is said to be* **informative** *if there are two types of A/C/G/T bases (gaps ignored) at the position in the alignment, and each type is supported by at least three reads.*

**Def**: *A read overlapping with the target read is* **inconsistent** *with the target if there are informative positions in the overlap and the read is not identical to the target read across all these positions; accordingly, the overlap between this and the target read is inconsistent.*

*Inconsistent reads theoretically originate from a haplotype different from the target read. Hifiasm only uses consistent reads to correct the target read.*
>

首先我们执行all-versus-all overlap alignment，然后进行矫正，对于待矫正的read称为Target read（简记为*T*），检查与*T*有overlap的所有reads的alignment情况，我们给出下面两个定义

Def_1: *T*中的某个位置称为**informative**，如果

(1)该位置的比对中存在两种碱基类型（不包括gap）

(2)每种碱基类型有至少三条read支持
![](/post_image/hifiasm/informative_pos.png)
__Fig.6 Informative position.__

Def_2；我们称一条read *R*（与*T*有overlap）相较于*T*是**inconsistent**，如果

(1)*R*与*T*的overlap中存在informative position

(2)二者informative position上的位置并不完全一致

![](/post_image/hifiasm/inconsistent_read.png)
__Fig.7 Inconsistent read.__

所以对于target read我们只使用inconsistent read来矫正，矫正的处理方法与Falcon类似，Hifiasm默认执行三轮矫正。

## Constructing phased assembly graphs

>
*The next step is to build the assembly string graph. In this graph, nodes represent oriented reads, and each edge between
two nodes represents the overlap between the corresponding two reads. Note that only consistent overlaps are used to build the graph. Because hifiasm builds the graph on top of nearly error-free reads and highly accurate haplotype phasing, the
produced assembly graph of hifiasm is simpler and cleaner than those of current assemblers for haploid genomes. However, for diploid genomes or polyploid genomes, its graph becomes more complicated as reads from different haplotypes are clearly separated out by phasing.*
>

下面进行构图，hifiasm采用的是string graph，即顶点表示reads，边表示顶点（reads）之间的overlap序列。注意，图中的边采用的都是consistent overlap。因为我们的目的是还原两个单倍型，所以我们认为consistent overlap连接的两个reads来自于同一条染色体。之所以能这样将read分开，主要是因为HiFi-read的高准确度，保证了我们可以进行这么精准的操作。当然了，因为同源染色体不一致的序列存在，所以图中会出现“bubble”的结构，如下图所示。注意到，如果我们真的从图中给的reads alignment的情况来构图，比下面的图肯定复杂很多，会增加很多边。所以其实，这里面进行了transitive reduction的操作，即如果图中存在$v_{1} \rightarrow v_{2}$,$v_{2} \rightarrow v_{3}$，且$v_{1} \rightarrow v_{3}$这样的情况，那么我们就删除边$v_{1} \rightarrow v_{3}$，一直进行下去，移除类似的边，直到图中不存在为止。这样才真正得到了下面的图。

![Bubble](/post_image/hifiasm/bubble.PNG)
__Fig.8 Bubble[^5].__

## Constructing a primary assembly

>
*The construction of the primary assembly aims to produce contigs including one set of haplotypes but may switch subregions
between haplotypes. In other words, each subregion in the primary assembly only comes from one haplotype, while the corresponding subregions of other haplotypes are removed as duplications.*
>

什么叫做primary assembly呢，这一步的想达到什么结果呢。前面已经已经粗略的了解primary assembly了，具体来讲，我们先尽可能的保证一组assembly的连续性，这个assembly的每个contig的每个子区域都可以确定完全来自于一种单倍型，然而contig的序列整体来看是在两个单倍型之间反复转换的（如图Fig.9）。同时，我们又保证，每个contig的子区域对应的染色体的区域，其相应的另一条同源染色体的该区域不会存在于primary assembly中。如下图Fig.10所示，上面生成的contig就被选入primary assembly集合中，而下面的序列作为同源区段，暂时被抛弃。

![Primary contig](/post_image/hifiasm/primary-contig.PNG)
__Fig.9 Primary contig[^5].__

![Primary assembly](/post_image/hifiasm/primary-assembly.PNG)
__Fig.10 Primary assembly[^6].__

理想状态，我们觉得hifiasm得到的phased graph应该是一个长链，然后中间依次有一些“Bubbles”，但是实际上图中还存在一些tips，就是一些单端的分支，类似下面这样，当然也可能不是只有一个点，可能很长。这些位置本来应该是一个bubble，但是因为某些原因破碎了。形成的原因，可能是这个区域的没有被测序read覆盖，也可能是别的原因形成的。

![Tips](/post_image/hifiasm/tips.png)
__Fig.11 Tips.__



# Results

[^1]:图片来源[De novo assembly of haplotype-resolved genomes with trio binning](https://www.nature.com/articles/nbt.4277)

[^2]:图片来源[Pacbio官网](https://www.pacb.com/smrt-science/smrt-sequencing/hifi-reads-for-highly-accurate-long-read-sequencing)

[^3]:图片来源[Phased diploid genome assembly with single-molecule real-time sequencing](https://www.nature.com/articles/nmeth.4035)

[^4]:图片来源[Single-cell template strand sequencing by Strand-seq enables the characterization of individual homologs](https://www.nature.com/articles/nprot.2017.029)

[^5]:图片来源[Haplotype-resolved de novo assembly using phased assembly graphs with hifiasm](https://www.nature.com/articles/s41592-020-01056-5)

[^6]:图片来源[Haplotype-resolved de novo assembly using phased assembly graphs with hifiasm-Supplementary information](https://static-content.springer.com/esm/art%3A10.1038%2Fs41592-020-01056-5/MediaObjects/41592_2020_1056_MOESM1_ESM.pdf)


